src/base58.cpp

```diff
@@ -136,7 +136,7 @@ bool DecodeBase58Check(const char* psz, std::vector<unsigned char>& vchRet)
     }
     // re-calculate the checksum, ensure it matches the included 4-byte checksum
     uint256 hash = Hash(vchRet.begin(), vchRet.end() - 4);
-    if (memcmp(&hash, &vchRet.end()[-4], 4) != 0) {
+    if (memcmp(&hash, vchRet.data() + vchRet.size() - 4, 4) != 0) {
```

Comment:
Yeah `vchRet.end()-4` would still be an iterator I believe, so you'd have to do some ugly thing like `&*(vchRet.end()-4)` to make it into a pointer iirc?